spec/features/admin/members/admin_search_member_spec.rb

```diff
@@ -0,0 +1,28 @@
+require 'rails_helper'
+
+feature 'Admin can search for users by email' do
+
+  signed_in_as(:admin) do
+    let!(:member) { FactoryGirl.create(:user, :member) }
+
+    before do
+      click_header_option("Dashboard")
+      click_sidemenu_option("Members")
+    end
+
+    scenario "Search with matching data" do
+      fill_in('User email search', :with => member.email)
+      click_button('SEARCH')
+      expect(page).to have_content(member.email)
+      expect(page).not_to have_content(current_user.email)
+    end
+
+    scenario "Search with non-matching data" do
+      fill_in('User email search', :with => current_user.email)
+      click_button('SEARCH')
+      expect(page).to have_content("No users matched that search")
+      expect(page).not_to have_content(current_user.email)
+    end
+  end
+
+end
```

Comment:
As far as I am concerned: It's just a good practise that has been widely adopted in a lot of small and big projects across the interwebs.

A lot of (node) projects for example purposely add a `.editorconfig` file http://editorconfig.org into their project that tells the IDE to add an extra line if supported.

There are a bunch of unix related reasons http://stackoverflow.com/questions/729692/why-should-text-files-end-with-a-newline, then there is github showing each file in a diff (because it's a unix thing) that doesn't have a new line at EOF like the screenshot I posted above.

>It's not about adding an extra newline at the end of a file, it's about not removing the newline that should be there.

Benefits:
- not being hassled by github about having removed a newline that should be there
- go with best practice

However, all the talking aside, at the end of the day it's going to be a personal preference. And it definitely is mine to have a newline at eof.